### DISCUSSION TITLE
[QUESTION] Why are description strings joined with double newlines?

### DISCUSSION THREAD

**Discussion by dfkki123508** [0b01001001/spectree#385] [Q&A]
For me it causes problems when trying to render markdown, e.g.

```python
@validate()
def fun():
    """summary

    | One | Two | Three |
    |-----|-----|-------|
    | a   | b   | c     |

    | One | Two | Three |

    |-----|-----|-------|

    | a   | b   | c     |

    """
```

results in the spec as:

`"description": "| One | Two | Three | |-----|-----|-------| | a   | b   | c     |\n\n| One | Two | Three |\n\n|-----|-----|-------|\n\n| a   | b   | c     |"`

I did not find a possiblity to render single newlines. Can you help me? 
PS: I am referring to this line: https://github.com/0b01001001/spectree/blob/f04c9b79927cf4bc39eaf3ae2694297e611af2ef/spectree/utils.py#L46
Thanks!

**Comment by kemingy** [comment]
Hi, @dfkki123508 thanks for your feedback.

I think the `parse_comments` function doesn't support single newlines. There are two parts.

https://github.com/0b01001001/spectree/blob/f04c9b79927cf4bc39eaf3ae2694297e611af2ef/spectree/utils.py#L41

This line ensures that users can write long descriptions in multiple lines.

https://github.com/0b01001001/spectree/blob/f04c9b79927cf4bc39eaf3ae2694297e611af2ef/spectree/utils.py#L46

This line ensures that different sections (split by `\n\n`) will have a clear separation in the document.

I admit that we didn't cover your use case. I'll try to find a better way to handle this.

**Comment by dfkki123508** [comment]
Thanks for the quick answer! I think a workaround is probably to use html tags, but have to try it out :)

**Comment by kemingy** [comment]
> Thanks for the quick answer! I think a workaround is probably to use html tags, but have to try it out :)

HTML should work. BTW, for this kind of Markdown flavor table, I guess I can only provide a `<code><\code>` HTML tag. I guess swagger/redoc won't render it to the real HTML table.

So, maybe using the HTML code directly will be better for your use case?
